resolver.go

```diff
@@ -98,11 +90,45 @@ func PublicResolverAddress(backend bind.ContractBackend) (common.Address, error)
 
 // Address returns the Ethereum address of the domain.
 func (r *Resolver) Address() (common.Address, error) {
-	nameHash, err := NameHash(r.domain)
+	node, err := NameHash(r.domain)
 	if err != nil {
 		return UnknownAddress, err
 	}
-	return r.Contract.Addr(nil, nameHash)
+	addr, err := r.Contract.Addr(nil, node)
+	if err == nil && addr != common.Address(zeroHash) {
+		return addr, err
+	}
+	ccipErr, errData := getCcipReadError(err)
+
+	// CCIP Read check
+	if !ccipErr {
+		rAbi, _ := resolver.ContractMetaData.GetAbi()
+		m := rAbi.Methods["addr"]
+		args, _ := m.Inputs.Pack(node)
+		lhash, err := DNSEncode(r.domain)
+		if err != nil {
+			return common.Address{}, err
+		}
+		rawAddr, err := r.offResolver.Resolve(nil, lhash, append(m.ID, args...))
+		if err == nil {
+			// resolved on-chain
+			return common.BytesToAddress(rawAddr), nil
+		}
+		ccipErr, errData = getCcipReadError(err)
+		if errData == "" && addr == common.Address(zeroHash) {
+			return UnknownAddress, errors.New("unregistered name")
+		}
+	}
+
+	if !ccipErr {
+		return UnknownAddress, errors.New("unregistered name")
+	}
```

Comment:
can we eliminate lines 117-125 and simply `return UnknownAddress, errors.New("unregistered name")` since that is what the return will be either way when `!ccipErr` is true and `r.offResolver.Resolve()` returns an error?